include/utils/mqtt_abstraction.hpp

```diff
@@ -64,6 +64,10 @@ class MQTTAbstraction {
     /// \copydoc MQTTAbstractionImpl::unsubscribe(const std::string&)
     void unsubscribe(const std::string& topic);
 
+    ///
+    /// \copydoc MQTTAbstractionImpl::clear_retained_topics()
+    void clear_retained_topics();
```

Comment:
TODO: maybe track the retained topics separately and move this function out of the MQTT abstraction